File: qtrading/strategy_config.lua
require('utils')

strategy_config = { --this object is parsed into a StrategyConfig object

    market = "MKT_DCE",
    feed_orig = "ORIG_HTD",
    feed_dest = "DEST_JINGYI",
    feed_type = "SRC_ACR_DCE_L2",
    trade_server = "auto",

    instruments = {
	"FUT_DCE_PP:201609"
    },

    sim_order_book = cons{
        AsiaQueueInstrOrderBookSpec,
        look_ahead_exchange = {
            enabled = true,
            verbose = true,
            use_fixed_look_ahead_time = true,
            fixed_look_ahead_time = Duration(.5), -- in seconds
            -- max_advance_duration = Duration(.5),
        }
    },

    -- other options can be added as needed
    start_hour = 10,
    start_minute = 15,
    start_second = 0,
    end_hour = 14,
    end_minute = 15,
    end_second = 0,

    shares = 100
}

File: qtrading/.build/default_trusty/alphaless/StrategyConfig.cc
#include "StrategyConfig.h"

namespace alphaless {

void StrategySettings::describe()
{
    Self::create()
        .param( "market"                     , &Self::market )
        .param( "feed_orig"                  , &Self::feed_orig )
        .param( "feed_dest"                  , &Self::feed_dest )
        .param( "feed_type"                  , &Self::feed_type )
        .param( "trade_server"               , &Self::trade_server )
        .param( "instruments"                , &Self::instruments )
        .param( "sim_order_book"             , &Self::sim_order_book )
        .param( "start_hour"                 , &Self::start_hour )
        .param( "start_minute"               , &Self::start_minute )
        .param( "start_second"               , &Self::start_second )
        .param( "end_hour"                   , &Self::end_hour )
        .param( "end_minute"                 , &Self::end_minute )
        .param( "end_second"                 , &Self::end_second )
        .param( "shares"                     , &Self::shares )
        ;
}


} // strategy


